src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/glide/MauiCustomViewTarget.java

```diff
@@ -34,10 +35,8 @@ public void onLoadFailed(@Nullable Drawable errorDrawable) {
             return;
         this.completed = true;
 
-        callback.onComplete(false,
-            errorDrawable,
-            () -> { } // TODO: explicitly release image
-        );
+        // trigger the callback out of this target
+        post(() -> callback.onComplete(false, errorDrawable, this::clear));
```

Comment:
Callbacks have to be on the `post` because if you try load a second image right after the first one, then you are still in the callback because the `TaskCompletionSource.SetResult()` continues on the same thread and thus all the code in C# that follows is effectively in the target still - which is not a Glide-supported operation.